rdt/hyper_transformer.py

```diff
@@ -253,25 +278,13 @@ def reverse_transform(self, data):
             pandas.DataFrame:
                 reversed data.
         """
-        if self.copy:
-            data = data.copy()
-
-        drop_columns = []
-        for column_name, transformer in self._transformers.items():
-            columns = self._get_columns(data, column_name)
-            if not columns.empty:
-                try:
-                    columns_data = data[columns].values
-                    reversed_data = transformer.reverse_transform(columns_data)
-                except AttributeError:
-                    # temporarily support both old and new style transformers
-                    rename = dict(zip(columns, transformer.output_columns))
-                    reversed_data = transformer.reverse_transform(data.rename(columns=rename))
-                    reversed_data = reversed_data[transformer.columns[0]].to_numpy()
-
-                data[column_name] = reversed_data
-                drop_columns.extend(set(columns) - {column_name})
-
-        data.drop(drop_columns, axis=1, inplace=True)
-
-        return data
+        for transformer in reversed(self._transformers_sequence):
```

Comment:
It looks like calling the `reverse_transform` methods of the transformers in reverse order produces the reversed columns in the wrong order (exactly the opposite than the one seen during fit). This seems unavoidable, but knowing that this happens I suggest that we add a way to restore the original order before returning.

NOTES:
- with this approach, `_temp_columns` may stop being needed too
- this `_subset` helper proposal can be used in other places of this module, and maybe somewhere else too. Should we create a helpers or utils module with these?

```python3
    @staticmethod
    def _subset(input_list, other_list, not_in=False):
        return [
            element
            for element in input_list
            if (element in other_list) ^ not_in
        ]

    def reverse_transform(self, data):
        unknown_columns = self._subset(data.columns, self._output_columns, not_in=True)
        for transformer in reversed(self._transformers_sequence):
            data = transformer.reverse_transform(data, drop=False)

        reversed_columns = self._subset(self._input_columns, data.columns)
        return reversed_data[unknown + reversed_columns]
```

This will keep all the unknown columns at the beginning of the table, and then add all the reversed columns at the end in the same order that was learned during `fit`. If the entire table is passed without any unknown columns, the result will be identical to the learned one.